src/configs/sastconfig.c4m

```diff
@@ -26,29 +26,38 @@ You can configure the following fields in the tool.semgrep object:
 
 semgrep_config_profile: The semgrep profile to use.  Defaults to 'auto'
 semgrep_format:         The output format flag to pass. Defaults to 'sarif'
-semgrep_metrics:        Whether to ping semgrep. 'on' or 'off'. Defaults to 'off'
+semgrep_metrics:        Whether to ping semgrep. 'on' or 'off'. Defaults to 'on' to be compatible with config=auto
 """
 }
 
+
+
 func find_semgrep(ignore) {
-  result := find_exe("semgrep", [])
+  semgrep_path := "/tmp/semgrep.env/bin/semgrep"
+  result := find_exe(semgrep_path, [])
 
   if result == "" {
-    trace("find_semgrep: Unable to find semgrep in $PATH")
+    trace("find_semgrep: Unable to find semgrep at" + result)
   } else {
     trace("found semgrep at: " + result)
   }
 }
 
 func install_semgrep(ignore) {
-  output, code := system("python3 -m pip install semgrep")
-  if code != 0 {
-    output, code := system("python -m pip install semgrep")
-  }
-  if code != 0 {
-    return false
+  trace("attempting to install semgrep to python3 venv")
+
+  # venv is only compatible with python3
+  output, code := system("python3 -m venv /tmp/semgrep.env")
+  if code == 0 {
+    output, code := system("/tmp/semgrep.env/bin/python3 -m pip install 'semgrep>=1.48'")
+    if code == 0 {
+      trace("semgrep successfully installed")
+      return true
+    }
   }
-  return true
+
+  trace("semgrep could not be installed")
+  return false
 }
 
 func get_semgrep_args(artifact) {
```

Comment:
Since we're touching this stuff should we make it more configurable?  Tho I guess a semgrep component might be better for that?

Reply:
I think a separate pr for that would be best